src/app/view/transport-call-creator/transport-call-creator.component.ts

```diff
@@ -312,10 +335,11 @@ export class TransportCallCreatorComponent implements OnInit {
         null,
       )
 
+      const publisherRoleSelected = this.transportCallFormGroup.controls.publisherRole.value;
       let date = this.transportCallFormGroup.controls.eventTimestampDate.value as Date;
       let time = this.transportCallFormGroup.controls.eventTimestampTime.value;
       // TODO: Let the user choose the role
-      timestamp.publisherRole = this.timestampMappingService.overlappingPublisherRoles(timestampDefinition)[0]
+      timestamp.publisherRole = !!publisherRoleSelected ? publisherRoleSelected : this.publisherRoles[0]; 
```

Comment:
Is the double `!!` necessary?  AFAICT, the LHS will already be forced in to a boolean context without it and the value will always come from the RHS (so the `any -> bool` type change itself seems redundant).

Reply:
This is resolved now